datetime/src/test/java/com/fasterxml/jackson/datatype/jsr310/ser/ZonedDateTimeSerTest.java

```diff
@@ -270,6 +273,64 @@ public void testSerializationAsStringWithZoneIdOn() throws Exception {
         assertEquals("The value is incorrect.", "\"" + DateTimeFormatter.ISO_ZONED_DATE_TIME.format(date) + "\"", value);
     }
 
+    @Test
+    public void testSerializationAsStringWithDefaultTimeZoneAndContextTimeZoneOnAndACustomFormatter() throws Exception {
+        ZonedDateTime date = ZonedDateTime.now(Z3);
+        // With a custom DateTimeFormatter without a ZoneId.
+        String value = MAPPER.registerModule(new SimpleModule().addSerializer(new ZonedDateTimeSerializer(FORMATTER_WITHOUT_ZONEID))).writer()
```

Comment:
As I mentioned in a comment, instead of constructing formatter, it'd make more sense to use `@JsonFormat` annotation or config overrides -- that's the expected usage. But this test is fine for now.